src/background/websocket-manager.ts

```diff
@@ -0,0 +1,361 @@
+          continue;
+        }
+        const { scripthash, network } = job;
+        await updateScriptHash(this.socket, network, scripthash, this.marinaStore);
+      } catch {
+        continue;
+      }
+    }
+  }
+
+  // stop the status worker
+  async stop(): Promise<void> {
+    if (!this.started) return;
+    this.started = false;
+    await this.socket.close('disconnection');
+  }
+
+  async forceUpdate(scripthash: string, network: NetworkString): Promise<void> {
+    await updateScriptHash(this.socket, network, scripthash, this.marinaStore);
+  }
+
+  async forceUpdateAccount(accountID: AccountID, network: NetworkString): Promise<void> {
+    const account = selectAccount(accountID)(this.marinaStore.getState());
+    if (!account) throw new Error(`Account ${accountID} not found`);
+    const id = await account.getWatchIdentity(network);
+    const addresses = await id.getAddresses();
+
+    const scriptsHashes = addresses
+      .map((a) => address.toOutputScript(a.confidentialAddress).toString('hex'))
+      .map(reverseAndHash);
+
+    // init the reducer state in order to link account to scripthash
+    for (const scripthash of scriptsHashes) {
+      try {
+        selectUtxosMapByScriptHash(network, scripthash)(this.marinaStore.getState());
+      } catch {
+        this.marinaStore.dispatch(addScriptHash(accountID, scripthash, network));
+      }
+    }
+
+    for (const scriptHash of scriptsHashes) {
+      await this.forceUpdate(scriptHash, network);
+    }
+  }
+}
+
+// utility function converting an output script hex to a scripthash (reversed sha256 of the script)
+// scripthash is used by electrum protocol to identify a script to watch
+export function reverseAndHash(script: string): string {
+  return crypto.sha256(Buffer.from(script, 'hex')).reverse().toString('hex');
+}
+
+// this function is triggered each time a new status is received for a scripthash
+// it fetches the new txs from history and updates the utxos and transactions state
+async function updateScriptHash(
+  ws: ElectrumWS,
+  network: NetworkString,
+  scriptHash: string,
+  store: Store
+): Promise<void> {
+  const historyResp = (await ws.request(
+    'blockchain.scripthash.get_history',
+    scriptHash
+  )) as History;
+  const historyDiff = selectHistoryDiff(historyResp)(store.getState());
+  const accountID = selectAccountIDByScriptHash(scriptHash, network)(store.getState());
+  const blindKeyGetter = getPrivateBlindKeyGetter(store, network);
+  const walletScripts = await getAllWalletScripts(store, network);
+
+  for (const { txID, height } of historyDiff.confirmedTxs) {
+    if (height <= 0) continue;
+    const blockHeaderHex = (await ws.request('blockchain.block.header', height)) as string;
+    const blockHeader = deserializeBlockHeader(blockHeaderHex);
+    const action = confirmTx(txID, blockHeader.timestamp, network);
+    store.dispatch(action);
+  }
+
+  // for each new tx which is not in the state, we fetch the transaction and unblind it
```

Comment:
we have plenty of operations here that require redux store getting data, which also may be I/O intensive. If we move to chrome.storage we can decrease the overhead maybe?

Like `blindKeyGetter`, can simply be replaced by passing the full master blinding key and derive it on the go or  if we cache script with the linked blidning key already dervie in storage, we can read sync from there?.